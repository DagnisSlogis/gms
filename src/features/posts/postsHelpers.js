// @flow
export const getSmallThumbnail = (link: string): string => {
  let re = /(?:\.([^.]+))?$/;
  let ext = re.exec(link)[0];
  link = link.replace(/\.[^/.]+$/, "");
  return link + "-390x410" + ext;
};

export const formatToLabel = (format: string): string => {
  switch (format) {
    case "image":
      return "- afi≈°a -";
    case "gallery":
      return "- galerija -";
    case "video":
      return "- video -";
    default:
      return "- raksts -";
  }
};
