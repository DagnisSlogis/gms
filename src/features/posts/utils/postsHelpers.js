// @flow
export const getSmallThumbnail = (link: string): string => {
  if (link) {
    let re = /(?:\.([^.]+))?$/;
    let ext = re.exec(link)[0];
    link = link.replace(/\.[^/.]+$/, "");
    return link + "-390x410" + ext;
  }
  return "";
};

export const formatToLabel = (format: string): string => {
  switch (format) {
    case "image":
      return "- afi≈°a -";
    case "gallery":
      return "- galerija -";
    case "video":
      return "- video -";
    default:
      return "- raksts -";
  }
};
