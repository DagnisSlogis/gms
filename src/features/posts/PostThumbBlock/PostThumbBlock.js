// @flow
import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import { getSmallThumbnail, formatToLabel } from "./../postsHelpers";

import {
  PostThumbImage,
  PostTypeBadge,
  PostTitle,
  PostBlock
} from "./PostThumbBlock.styles";

type Props = {
  post: Object,
  onClick: Function
};

export default class PostThumbBlock extends PureComponent<Props> {
  static propTypes = {
    post: PropTypes.object.isRequired,
    onClick: PropTypes.func.isRequired
  };

  featuredImg = () =>
    this.props.post._embedded["wp:featuredmedia"]["0"].source_url;

  render = () => (
    <PostBlock onClick={event => this.props.onClick(event, this.props.post)}>
      <PostThumbImage
        image={getSmallThumbnail(this.featuredImg())}
        height={305}
      />
      <PostTypeBadge>{formatToLabel(this.props.post.format)}</PostTypeBadge>
      <PostTitle>{this.props.post.title.rendered}</PostTitle>
    </PostBlock>
  );
}
