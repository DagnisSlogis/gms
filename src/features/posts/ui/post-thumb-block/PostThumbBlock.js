// @flow
import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import { getSmallThumbnail, formatToLabel } from "../../utils/postsHelpers";

import {
  PostThumbImage,
  PostTypeBadge,
  PostTitle,
  PostBlock
} from "./PostThumbBlock.styles";

type Props = {
  post: Object,
  onClick: Function
};

export default class PostThumbBlock extends PureComponent<Props> {
  static propTypes = {
    post: PropTypes.object.isRequired,
    onClick: PropTypes.func.isRequired
  };

  featuredImg = () => {
    console.log(this.props.post._embedded["wp:featuredmedia"]);
    return this.props.post._embedded["wp:featuredmedia"]
      ? this.props.post._embedded["wp:featuredmedia"]["0"].source_url
      : null;
  }

  render = () => (
    <PostBlock onClick={event => this.props.onClick(event, this.props.post)}>
      <PostThumbImage
        image={getSmallThumbnail(this.featuredImg())}
        height={305}
      />
      <PostTypeBadge>{formatToLabel(this.props.post.format)}</PostTypeBadge>
      <PostTitle>{this.props.post.title.rendered}</PostTitle>
    </PostBlock>
  );
}
