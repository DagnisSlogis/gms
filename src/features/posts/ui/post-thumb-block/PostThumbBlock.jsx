import React from "react";
import PropTypes from "prop-types";
import SmartGallery from 'react-smart-gallery';

import { getGalleryImages } from 'utils/contentUtils';

import { getSmallThumbnail, formatToLabel } from "../../utils/postsHelpers";
import {
  PostThumbImage,
  PostTypeBadge,
  PostTitle,
  PostBlock,
  GalleryThumb,
} from "./PostThumbBlock.styles";


export const PostThumbBlock = ({
  post,
  post: {
    _embedded,
    format,
    content: { rendered: content },
    title: { rendered: title },
  },
  onClick,
}) => {
  const featuredImg = () =>
    _embedded["wp:featuredmedia"] ?
      _embedded["wp:featuredmedia"]["0"].source_url :
      null;

  const renderPostThumb = () => {
    if (format !== 'gallery') {
      return (
        <PostThumbImage
          image={getSmallThumbnail(featuredImg())}
          height={305}
        />
      );
    }

    return (
      <GalleryThumb>
        <SmartGallery
          images={getGalleryImages(content)}
          width='100%'
          height={305}
        />
      </GalleryThumb>
    );
  }

  return (
    <PostBlock onClick={(e) => onClick(e, post)}>
      {renderPostThumb()}
      <PostTypeBadge>{formatToLabel(format)}</PostTypeBadge>
      <PostTitle>{title}</PostTitle>
    </PostBlock>
  )
}

PostThumbBlock.propTypes = {
  post: PropTypes.object.isRequired,
  onClick: PropTypes.func.isRequired
};
