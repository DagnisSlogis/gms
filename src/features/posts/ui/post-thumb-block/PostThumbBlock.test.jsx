// jest PostThumbBlock
import React from "react";
import { shallow } from "enzyme";
import { PostThumbBlock } from "./PostThumbBlock";

import { getSmallThumbnail, formatToLabel } from "../postsHelpers";
jest.mock("../postsHelpers", () => ({
  getSmallThumbnail: jest.fn(),
  formatToLabel: jest.fn()
}));

describe("<PostThumbBlock/>", () => {
  describe("Has post in post prop", () => {
    const post = {
      id: 1,
      format: "video",
      title: {
        rendered: "Title 1"
      },
      _embedded: {
        "wp:featuredmedia": {
          "0": {
            source_url: "test"
          }
        }
      }
    };

    it("renders correctly", () => {
      const wrapper = shallow(<PostThumbBlock post={post} />);
      expect(wrapper).toMatchSnapshot();
      expect(getSmallThumbnail.mock.calls.length).toBe(1);
      expect(wrapper.length).toEqual(1);
      expect(formatToLabel.mock.calls.length).toBe(1);
    });
  });
});
