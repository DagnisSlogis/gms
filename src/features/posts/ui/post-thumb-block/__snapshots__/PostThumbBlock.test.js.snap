// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostThumbBlock/> Has post in post prop renders correctly 1`] = `
<styled.div>
  <styled.img />
  <styled.span />
  <styled.h3>
    Title 1
  </styled.h3>
</styled.div>
`;
